corecfg/proxy.go

```diff
@@ -29,9 +29,9 @@ import (
 )
 
 var proxyConfigKeys = map[string]bool{
-	"http":  true,
-	"https": true,
-	"ftp":   true,
+	"http_proxy":  true,
```

Comment:
This looks OK upon code inspection. I second @pedronis' comment though. It should be a separate commit with a description of what is going on. It seems this was just a bug all along and we didn't notice because of a bug in the `updateKeyValueStream`. With that fix this one surfaced.